packages/react-components/react-motion-components-preview/library/src/choreography/Stagger/utils/stagger-calculations.ts

```diff
@@ -0,0 +1,94 @@
+import type { StaggerProps } from '../stagger-types';
+import { DEFAULT_ITEM_DELAY, DEFAULT_ITEM_DURATION } from './constants';
+
+/**
+ * Calculate the total stagger duration—from the moment stagger begins
+ * until the final item's animation completes.
+ *
+ * Uses the formula:
+ *   max(0, delay * (count - 1) + itemDuration)
+ *
+ * @param params.count        Total number of items to stagger
+ * @param params.delay        Milliseconds between the start of each item
+ * @param params.itemDuration Milliseconds each item's animation lasts (default 0)
+ * @returns                   Total duration in milliseconds (never negative)
+ */
+export function getStaggerTotalDuration({
```

Comment:
<img width="768" height="211" alt="image" src="https://github.com/user-attachments/assets/f6de3d1d-72f2-4fb8-b68b-a327170ed805" />

JSDoc there and in other places is invalid